src/plugins/intel_gna/frontend/quantization.cpp

```diff
@@ -200,262 +54,228 @@ float ScaleFactorForQuantization(void *ptr_float_memory, float target_max, size_
+            value = value * ptr_weight_scale_factor + rounding_value;
+
+            int16_t* ptr_weight_16 = ptr_int_weights + (row * data.num_columns + col);
 
-            value = ptr_float_weights[row * num_columns + col] * (*ptr_weight_scale_factor / ptr_int_biases[row].multiplier) + rounding_value;
-            if (value > 127.0) {
-                *ptr_weight_8 = 127;
+            if (value > std::numeric_limits<int16_t>::max()) {
+                *ptr_weight_16 = std::numeric_limits<int16_t>::max();
                 num_saturate++;
-            } else if (value < -128.0) {
-                *ptr_weight_8 = -128;
+            } else if (value < std::numeric_limits<int16_t>::min()) {
+                *ptr_weight_16 = std::numeric_limits<int16_t>::min();
                 num_saturate++;
             } else {
-                *ptr_weight_8 = (int8_t) value;
+                *ptr_weight_16 = (int16_t)value;
             }
         }
-        for (uint32_t col = num_columns; col < num_columns_padded; col++) {
-            int8_t *ptr_weight_8 = ptr_int_weights + (row * num_columns_padded + col);
-            *ptr_weight_8 = 0;
-        }
     }
-    for (uint32_t row = num_rows; row < num_rows_padded; row++) {
-        for (uint32_t col = 0; col < num_columns_padded; col++) {
-            int8_t *ptr_weight_8 = ptr_int_weights + (row*num_columns_padded + col);
-            *ptr_weight_8 = 0;
-        }
-        ptr_int_biases[row].multiplier = 0;
+
+    if (num_saturate > 0) {
+        QUANTWARNING("Warning:  %d / %d saturations in QuantizeAffine16()\n",
+            num_saturate,
+            data.num_rows * data.num_columns + data.num_rows);
     }
+}
 
-    // bias value of the bas will be only used when input bias provided
-    if (ptr_float_biases != nullptr) {
-        for (uint32_t j = 0; j < num_rows; j++) {
-            float rounding_value = (ptr_float_biases[j] > 0) ? 0.5f : -0.5f;
-            float value = ptr_float_biases[j] * *ptr_output_scale_factor + rounding_value;
+template<>
+void BiasQuantizationCallback<int8_t>::quantize() const {
+    // Stub
+}
+
+template<>
+void BiasQuantizationCallback<int16_t>::quantize() const {
+    // Stub
+}
+
+template<>
+void BiasQuantizationCallback<int32_t>::quantize() const {
+    uint32_t num_saturate = 0;
+
+    // case for element wise layer
+    if (ptr_float_biases != nullptr && ptr_int_biases != nullptr) {
+        for (size_t row = 0; row < data.num_rows; row++) {
+            float rounding_value = (ptr_float_biases[row] > 0) ? 0.5f : -0.5f;
+            float value = ptr_float_biases[row] * ptr_output_scale_factor + rounding_value;
             if (value > 2147483647.0) {
-                ptr_int_biases[j].bias = 2147483647L;
+                ptr_int_biases[row] = 2147483647L;
                 num_saturate++;
             } else if (value < -2147483648.0) {
-                ptr_int_biases[j].bias = -2147483648LL;
+                ptr_int_biases[row] = -2147483648LL;
```

Comment:
`-2147483648LL` is it OK in previous case there is `L` ? Can this values be defined or use `std::numeric_limits`?

Same comment for block L:263-274